test/tools/ossfuzz/protomutators/YulProtoMutator.cpp

```diff
@@ -107,9 +107,7 @@ template <typename T>
 T YPM::EnumTypeConverter<T>::validEnum(unsigned _seed)
 {
 	auto ret = static_cast<T>(_seed % (enumMax() - enumMin() + 1) + enumMin());
-	if constexpr (std::is_same_v<std::decay_t<T>, FunctionCall_Returns>)
```

Comment:
It looks like `FunctionCall` "things" were removed, how does it work now?

Reply:
Actually, a portion of the `FunctionCall` specification was altered. Earlier, the `FunctionCall` spec included the number of returned values and a pseudo-random index to select a previously declared function. The encoded logic was that, the proto converter selects a random (previously declared) function from the list of functions.

Say, the function signatures of previously declared functions are as follows:

```
f(a, b, c) -> d, e
g(a) -> b
h()
```

Now, the old spec worked like so. The fuzzer supplies
  - a random index say `11345` to select a function (we perform a modulo of number of previously declared functions)
  - the number of returned values desired (e.g., 0, 1, 2,3, 4 categorised as no return, single return, and multi return)

The converter would first perform a modulo on the index `11345 % 3 = 2` and select `h()` (see list of previously declared functions). In the next step, the converter would check if the function signature of the selected function matches the desired number of return values (no return, single, multiple).

In this case, although the converter found a function `f()` that could be used to make a function call, it may not have been inserted in the test case if the signature of the function did not match the number of return values specified by the fuzzer e.g., 1, 2, 3, 4.

I realised that this logic meant that fewer function calls are inserted into test cases generated by the fuzzer. Since there was no need to have the fuzzer additional specific number of return values of a function, I removed it to have a greater incidence of function calls.

I think a greater probability of function calls is useful to test optimiser steps like function inliner, and in general steps that simplify control logic.

In this PR, I removed the function selector index and number of function returns from the function call specification. Instead, the random selection is based on a monotonically increasing counter that already existed. Furthermore, once the function is selected, it is called in the generated test case.

Reply:
Ok, thanks!